pgrx/src/spi/query.rs

```diff
@@ -228,6 +233,9 @@ impl<'conn: 'stmt, 'stmt> Query<'conn> for &'stmt PreparedStatement<'conn> {
         limit: Option<libc::c_long>,
         arguments: Self::Arguments,
     ) -> SpiResult<SpiTupleTable<'conn>> {
+        if self.mutating {
+            Spi::mark_mutable();
+        }
```

Comment:
My understanding is Postgres will do this for us, and we can instead use our knowledge to actually skip some FFI overhead on our side, so we should, instead of this, make our bool passed to Postgres be the expr `self.mutating || Spi::is_xact_still_immutable()` on line 260.